### DISCUSSION TITLE
Why drf has decided to use metaclass for BasePermision?

### DISCUSSION THREAD

**Discussion by 7xMind** [encode/django-rest-framework#8544] [Question & Answer] [answered]
I need some expert advice on why Django rest framework has decided to go with metaclasses here.

```
class BasePermission(metaclass=BasePermissionMetaclass):
    """
    A base class from which all permission classes should inherit.
    """

    def has_permission(self, request, view):
        """
        Return `True` if permission is granted, `False` otherwise.
        """
        return True

    def has_object_permission(self, request, view, obj):
        """
        Return `True` if permission is granted, `False` otherwise.
        """
        return True
```

 Would a simple inheritance or leveraging `__new__`, or `__init__subclass__` be simpler and still achieve the same thing?

**Comment by lovelydinosaur** [comment, marked as answer]
It is lost in the mists of time. (At least to me)
Perhaps some future digital archeologist will be able to answer this question for us. 🕶

**Reply by 7xMind** [reply]
So, is it safe to say that the precise same behavior can be achieved by moving away from metaclasses to leveraging `__new__` or `__init__subclass__`?
